pkg/spiceerrors/assert_on.go

```diff
@@ -0,0 +1,12 @@
+//go:build ci
+// +build ci
+
+package spiceerrors
+
+import "fmt"
+
+func DebugAssert(condition func() bool, format string, args ...any) {
+	if !condition() {
```

Comment:
why not have the condition fn return an error?

then you can do `DebugAssert(func() {return fmt.Errorf("whatever")}`

Reply:
I personally think this reads better: it ensures that the conditional is a boolean and we could take other actions too (like log)

Reply:
But you won't be able to return messages that depend on computation in the assert fn

Reply:
That's fine though; its a debug assertion